pandas/core/arrays/categorical.py

```diff
@@ -2370,12 +2370,14 @@ def replace(self, to_replace, value, inplace: bool = False):
 
     # ------------------------------------------------------------------------
     # String methods interface
-    def _str_map(self, f, na_value=np.nan, dtype=np.dtype("object")):
+    def _str_map(self, f, na_value=np.nan, dtype=None):
         # Optimization to apply the callable `f` to the categories once
         # and rebuild the result by `take`ing from the result with the codes.
         # Returns the same type as the object-dtype implementation though.
         from pandas.core.arrays import PandasArray
 
+        if dtype is None:
+            dtype = np.dtype("object")
```

Comment:
why is this worthwhile?  id rather just annotate _str_map to have `dtype: np.dtype = np.dtype("object")`

Reply:
I don't like it either :) If `dtype` is never changes, it would be better to add an ignore statement.

Reply:
Using ignore statement now